development/tools/cmd/synchronizer/syncomponent/component.go

```diff
@@ -0,0 +1,91 @@
+package syncomponent
+
+import (
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Component represents the element for which version will be checked
+type Component struct {
+	Name       string
+	Path       string
+	GitHash    string
+	CommitDate string
+	outOfDate  bool
+	Versions   []*ComponentVersions
+}
+
+// ComponentVersions is a part of Component which includes information about version
+type ComponentVersions struct {
+	VersionPath   string
+	Version       string
+	ModifiedFiles []string
+	outOfDate     bool
+}
+
+// NewSynComponent returns new Component element
+func NewSynComponent(componentPath string, versionPaths []string) *Component {
+	componentVersions := []*ComponentVersions{}
+	for _, versionPath := range versionPaths {
+		componentVersion := &ComponentVersions{VersionPath: versionPath}
+		componentVersions = append(componentVersions, componentVersion)
+	}
+
+	nameElements := strings.Split(componentPath, "/")
+	nameDash := nameElements[len(nameElements)-1]
+	name := strings.Replace(nameDash, "-", "_", -1)
+
+	return &Component{
+		Name:     name,
+		Path:     componentPath,
+		Versions: componentVersions,
+	}
+}
+
+// CheckIsOutOfDate determines whether a given component is not expired
+func (c *Component) CheckIsOutOfDate(outOfDateDays int) {
+	days := daysDelta(c.CommitDate, time.Now().Unix())
```

Comment:
this code is problematic in case we update component every day but still don't update values.yaml

Try such approach:
````
if usedVersion == currentVersion {
 // do nothing
} else {
  commitsOrderedByDate = get(...)
  remove commits from last 3 days
 compare commitsOrderedByDate[0] and usedVersion
}
```